backend/controllers/taskController.js

```diff
@@ -88,104 +93,80 @@ const getTasksByDay = asyncHandler(async (req, res) => {
     } catch (error) {
       res.status(400);
       throw new Error("Invalid Date Input");
-
     }
   }
 
-  const tasks = await Task.find({
-    user_id: ObjectId(userId),
-    startDate: {
-      // $gte: startDate,
-      $gte: new Date(startDate),
+  const tasks = await Task.find(
+    {
+      user_id: ObjectId(userId),
+      startDate: {
+        // $gte: startDate,
+        $gte: new Date(startDate),
+      },
+      endDate: {
+        $lte: new Date(endDate),
+      },
+      isStarted: true,
     },
-    endDate: {
-      $lte: new Date(endDate),
-    },
-    isStarted: true,
-  }, (err, docs) => {
-    if(err) {
-      res.status(500);
-      throw new Error(`Could not fetch doc ${docs}`)
+    (err, docs) => {
```

Comment:
I think for consistency, we should also change this to using a `.then` and `.catch` clause

Reply:
This change can be done in the getTasks PR

Reply:
Nevermind, this can actually be kept here since the change is already in the get tasks PR

Reply:
If anything, I would just copy over the function from the get tasks PR and just make sure that it's the same (so we're not overriding it with this implementation rather than the one that we need)